### DISCUSSION TITLE
per workspace layout?

### DISCUSSION THREAD

**Discussion by tve** [hyprwm/Hyprland#3477] [Q&A] [answered]
Looking through the wiki it seems that it's not possible to choose different layouts in different workspaces, is that correct?

**Comment by vaxerski** [comment, marked as answer]
it is correct

**Reply by tve** [reply]
bwaaaa! 😢

**Comment by DreamMaoMao** [comment]
@vaxerski 

Is it possible to achieve dynamic layout switching by writing plugins?

**Reply by vaxerski** [reply]
uhh you can switch layouts dynamically by just changing the `general:layout` var.

If you mean per-workspace layouts, no.

**Reply by DreamMaoMao** [reply]
ok thanks

**Reply by justyn** [reply]
@DreamMaoMao perhaps it is possible to write a new layout plugin that merges the functionality of the different layouts you want into one "mega layout", with rules that alter according to the current workspace?

**Reply by DreamMaoMao** [reply]
@justyn 
It's totally achievable, but it's not what I want. Most people probably won't use it, so I won't implement it

**Reply by zakk4223** [reply]
I ended up writing a plugin that does this. https://github.com/zakk4223/hyprWorkspaceLayouts
